### PR TITLE
Fix poll pkg bug for buildconfig resources

### PR DESCRIPTION
**Pull request by pacevedom**
When there is more than one `buildconfig` resource in the same namespace, poll package queries and waits for all builds, ignoring the build owner, and in turn, possibly waiting on not owned resources. Also fixes waiting for not yet created build pods.

### CODE REVIEW

File: pkg/poll/poll.go [hunk 1]
```diff
@@ -411,14 +411,27 @@ func ForBuild(obj *unstructured.Unstructured) error {
...
-		callback := makeStatusCallback(&build, "Complete", "status", "phase")
-		if err := ForResourceFullAvailability(&build, callback); err != nil {
+	var build *unstructured.Unstructured
+	for _, b := range builds.Items {
+		slice, _, err := unstructured.NestedSlice(b.Object, "metadata", "ownerReferences")
+		if err != nil {
 			return err
 		}
+		for _, element := range slice {
+			if name, ok := element.(map[string]interface{})["name"]; ok && name == obj.GetName() {
```
Comment (qbarrand) [inline pkg/poll/poll.go:421]: If the key does not exist in the map, the value will be the empty string. I'd say `ok` is redundant here.
```suggestion
			if element.(map[string]interface{})["name"]== obj.GetName() {
```

**Review by qbarrand — COMMENTED** [review]
Besides one minor suggestion, lgtm 🙂

**Comment by qbarrand** [comment]
Would be great if you could pull the tests from https://github.com/openshift-psap/special-resource-operator/pull/241 as well.

**Comment by pacevedom** [comment]
> Would be great if you could pull the tests from [openshift-psap#241](https://github.com/openshift-psap/special-resource-operator/pull/241) as well.

@qbarrand Would rather do that in a separate commit, the original pkg is quite different after the tests were added.

**Pull request merged**